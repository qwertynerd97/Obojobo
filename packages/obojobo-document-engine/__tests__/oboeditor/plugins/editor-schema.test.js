import { CHILD_REQUIRED, CHILD_TYPE_INVALID } from 'slate-schema-violations'

import EditorSchema from 'src/scripts/oboeditor/plugins/editor-schema'

describe('EditorSchema', () => {
	test('matcher builds a list of all possible node types', () => {
		expect(EditorSchema.schema).toMatchSnapshot()
	})
	test('schema.normalize fixes required children in editor', () => {
		const change = {
			insertNodeByKey: jest.fn()
		}

		EditorSchema.schema.document.normalize(change, {
			code: CHILD_REQUIRED,
			node: { key: 'mockKey' },
			child: null,
			index: 0
		})

		expect(change.insertNodeByKey).toHaveBeenCalled()
	})

	test('schema.normalize fixes invalid children in editor', () => {
		const change = {
			wrapBlockByKey: jest.fn()
		}

		EditorSchema.schema.document.normalize(change, {
			code: CHILD_TYPE_INVALID,
			node: { key: 'mockKey' },
			child: { object: 'text', key: 'mockChild' },
			index: 0
		})

		expect(change.wrapBlockByKey).toHaveBeenCalled()
	})
})
