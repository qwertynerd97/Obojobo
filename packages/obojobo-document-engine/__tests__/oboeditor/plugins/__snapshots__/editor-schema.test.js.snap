// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSchema matcher builds a list of all possible node types 1`] = `
Object {
  "document": Object {
    "nodes": Array [
      Object {
        "match": Array [
          Object {
            "type": "ObojoboDraft.Chunks.ActionButton",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Break",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Code",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Figure",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Heading",
          },
          Object {
            "type": "ObojoboDraft.Chunks.HTML",
          },
          Object {
            "type": "ObojoboDraft.Chunks.IFrame",
          },
          Object {
            "type": "ObojoboDraft.Chunks.List",
          },
          Object {
            "type": "ObojoboDraft.Chunks.MathEquation",
          },
          Object {
            "type": "ObojoboDraft.Chunks.MCAssessment",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Table",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Text",
          },
          Object {
            "type": "ObojoboDraft.Chunks.YouTube",
          },
          Object {
            "type": "ObojoboDraft.Chunks.QuestionBank",
          },
          Object {
            "type": "ObojoboDraft.Chunks.Question",
          },
          Object {
            "type": "ObojoboDraft.Sections.Assessment",
          },
          Object {
            "type": "ObojoboDraft.Pages.Page",
          },
        ],
        "min": 1,
      },
    ],
    "normalize": [Function],
  },
}
`;
